CDTDatastore/CDTReplicator/CDTAbstractReplication.h

```diff
@@ -146,6 +146,16 @@ typedef NS_ENUM(NSInteger, CDTReplicationErrors) {
  * @return an initialsed instance of CDTAbstractReplication.
  */
 - (instancetype)initWithUsername:(nullable NSString*)username password:(nullable NSString*)password;
+
+/**
+ * Initalises the abstract replication, using an IAM API key to authenticate.
+ * See https://console.bluemix.net/docs/services/Cloudant/guides/iam.html#ibm-cloud-identity-and-access-management
+ * for more information about IAM.
+ * @param iamApiKey The IAM API key.
+ * @return an initialsed instance of CDTAbstractReplication.
+ */
+- (instancetype)initWithIamApiKey:(NSString *)iamApiKey;
```

Comment:
Although I failed to account for the fully lowercase versions `iam` the guidelines do say one of the exceptions to method naming is to start with an acronym, so method names starting `IAM` should also be acceptable. I've stopped marking instances, I suggest a global find refactor to a consistent style :)